### ISSUE TITLE
Module level user-defined types which are based on constants, have type checking errors

### ISSUE THREAD

**Issue by SandaruJayawardana** [ballerina-platform/ballerina-lang#33890]
**Description:**
$title. This happens when the constants are initialized with expressions.

**Steps to reproduce:**
```ballerina
const int X = 1 + 7;
X y = 2; // No errors!
```

**Comment by SandaruJayawardana** [comment]
Related issue.
```ballerina
public enum Color {
    RED = "r" + "ed"
}

public function main() {
    Color a = "RED"; // No errors!
}
```

**Comment by pcnfernando** [comment]
Once fixed tests added in https://github.com/ballerina-platform/ballerina-lang/pull/34806/ should be enabled

**Comment by gimantha** [comment]
Fixed in 2201.10.0
